pkg/auth/dependency/userverify/verify_flag_test.go

```diff
@@ -10,148 +11,217 @@ import (
-					"email": "test+1@example.com",
-					"phone": "+85299999999",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, false)
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{}),
-				makePrincipals(map[string]string{
-					"phone": "+85299999999",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, false)
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{}),
-				makePrincipals(map[string]string{
-					"username": "test",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, false)
+		Convey("When no keys to verify", func() {
+			Convey("should check empty login IDs", func() {
+				So(isUserVerified(verifyRequest{
+					LoginIDs:          map[string]string{},
+					VerifiedLoginIDs:  []string{},
+					VerifyLoginIDKeys: []string{},
+				}), ShouldResemble, verifyResult{All: false, Any: false})
+			})
+			Convey("should check single email login ID", func() {
+				So(isUserVerified(verifyRequest{
+					LoginIDs:          map[string]string{"email": "a@example.com"},
+					VerifiedLoginIDs:  []string{},
+					VerifyLoginIDKeys: []string{},
+				}), ShouldResemble, verifyResult{All: false, Any: false})
+			})
+			Convey("should check multiple email login ID", func() {
+				So(isUserVerified(verifyRequest{
+					LoginIDs:          map[string]string{"email": "a@example.com b@example.com"},
+					VerifiedLoginIDs:  []string{},
+					VerifyLoginIDKeys: []string{},
+				}), ShouldResemble, verifyResult{All: false, Any: false})
+			})
 		})
 
-		Convey("With criteria = any", func() {
-			var isVerified bool
-			criteria := config.UserVerificationCriteriaAny
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{
-					"test+1@example.com",
-				}),
-				makePrincipals(map[string]string{
-					"email": "test+1@example.com",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, true)
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{
-					"test+1@example.com",
-				}),
-				makePrincipals(map[string]string{
-					"email": "test+1@example.com",
-					"phone": "+85299999999",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, true)
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{}),
-				makePrincipals(map[string]string{
-					"phone": "+85299999999",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, false)
-
-			isVerified = IsUserVerified(
-				makeAuthInfo([]string{}),
-				makePrincipals(map[string]string{
-					"username": "test",
-				}),
-				criteria, verifyConfigs,
-			)
-			So(isVerified, ShouldEqual, false)
+		Convey("When need to verify email", func() {
+			Convey("should check empty login IDs", func() {
+				So(isUserVerified(verifyRequest{
+					LoginIDs:          map[string]string{},
+					VerifiedLoginIDs:  []string{},
+					VerifyLoginIDKeys: []string{"email"},
+				}), ShouldResemble, verifyResult{All: true, Any: false})
```

Comment:
It seems that `All: false` is more expected.

> User verification is enabled if VerifyLoginIDKeys is non-empty

> If user verification is disabled, All and Any must be false.

> When user verification is enabled, All is true if the login ids that requires verification are all verified and the set is non-empty

> When user verification is enabled, Any is true if the login ids that requires verification has at least one verified.

Do you think the above definition is better in the sense of usability?